includes/class-edd-logging.php

```diff
@@ -217,15 +217,15 @@ public function insert_log( $log_data = array(), $log_meta = array() ) {
 
 			// Use the right log fetching function based on the type of log this is.
 			if ( 'edd_add_api_request_log' === $insert_method ) {
-				$log = edd_get_file_download_log( $log_id );
+				$add_meta = 'edd_add_api_request_log_meta';
 			} elseif ( 'edd_add_file_download_log' === $insert_method ) {
-				$log = edd_get_file_download_log( $log_id );
+				$add_meta = 'edd_add_file_download_log_meta';
 			} else {
-				$log = edd_get_log( $log_id );
+				$add_meta = 'edd_add_log_meta';
 			}
 
 			foreach ( (array) $log_meta as $key => $meta ) {
-				$log->add_meta( sanitize_key( $key ), $meta );
+				$add_meta( $log_id, sanitize_key( $key ), $meta );
```

Comment:
While there's no real risk of this causing any error as-is, just to be super safe.. Can you wrap the whole `foreach` statement in:

```php
if ( is_callable( $add_meta ) ) {
    // foreach in here
}
```

Also, for clarity, maybe rename `$add_meta` to `$add_meta_function`. `$add_meta` feels like it would contain a boolean ("yes, I want to add meta").